domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt

```diff
@@ -0,0 +1,289 @@
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.RuleSpec
+import org.oppia.app.model.State
+import org.oppia.app.model.StringList
+import org.oppia.app.model.SubtitledHtml
+import org.oppia.util.data.DataProviders
+
+const val TEST_EXPLORATION_ID_0 = "test_exp_id_0"
+const val TEST_EXPLORATION_ID_1 = "test_exp_id_1"
+private const val WELCOME_EXPLORATION_DATA_PROVIDER_ID = "WelcomeExplorationDataProvider"
+private const val ABBOUT_OPPIA_EXPLORATION_DATA_PROVIDER_ID = "AboutOppiaExplorationDataProvider"
+
+/** Controller for retrieving an exploration. */
+@Singleton
+class ExplorationDataController @Inject constructor(private val context: Context,
+                                                    private val dataProviders: DataProviders
+) {
+
+  private val welcomeExplorationDataProvider =
+    dataProviders.createInMemoryDataProviderAsync(
+      WELCOME_EXPLORATION_DATA_PROVIDER_ID, this::retrieveWelcomeExplorationAsync)
+  private val aboutOppiaExplorationDataProvider =
+    dataProviders.createInMemoryDataProviderAsync(
+      ABBOUT_OPPIA_EXPLORATION_DATA_PROVIDER_ID, this::retrieveAbboutOppiaExplorationAsync)
+
+  /**
+   * Returns an  [Exploration] given an ID.
+   */
+  fun getExplorationById(ID: String): LiveData<AsyncResult<Exploration>>? {
+    if (ID == TEST_EXPLORATION_ID_0) {
+      return dataProviders.convertToLiveData(welcomeExplorationDataProvider)
+    }
+    if (ID == TEST_EXPLORATION_ID_1) {
+      return dataProviders.convertToLiveData(aboutOppiaExplorationDataProvider)
+    }
+    return null
+  }
+
+  @Suppress("RedundantSuspendModifier") // DataProviders expects this function to be a suspend function.
+  private suspend fun retrieveWelcomeExplorationAsync(): AsyncResult<Exploration> {
+    return AsyncResult.success(createExploration("welcome.json"))
+  }
+
+  @Suppress("RedundantSuspendModifier") // DataProviders expects this function to be a suspend function.
+  private suspend fun retrieveAbboutOppiaExplorationAsync(): AsyncResult<Exploration> {
+    return AsyncResult.success(createExploration("about_oppia.json"))
+  }
+
+  // Returns an exploration given an assetName
+  private fun createExploration(assetName: String): Exploration {
+    val explorationObject = loadJSONFromAsset(assetName)?: return Exploration.getDefaultInstance()
+    return Exploration.newBuilder()
+      .setTitle(explorationObject?.getString("title"))
+      .setLanguageCode(explorationObject?.getString("language_code"))
+      .setInitStateName(explorationObject?.getString("init_state_name"))
+      .setObjective(explorationObject?.getString("objective"))
+      .putAllStates(createStatesFromJsonObject(explorationObject?.getJSONObject("states")))
+      .build()
+  }
+
+  // Returns a JSON Object if it exists, else returns null
+  private fun getJSONObject(parentObject: JSONObject, key: String): JSONObject? {
+    try {
+      return parentObject.getJSONObject(key)
+    } catch(e: JSONException) {
```

Comment:
Suggest throwing this instead and catching it in the createExploration method, then passing that back to the UI via AsyncResult (this is more or less the intent of AsyncResult: it provides pending, value, and error channels to each of which the UI can respond).

Reply:
In this particular case I actually want to catch and handle it because there are situations where the object can be null, however it makes sense to do this in loadJsonFromAsset, done it there.